browser/components/dataselector/ProjectListItem.js

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react';
+import {CopyToClipboard} from 'react-copy-to-clipboard';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import styled from "styled-components";
+import Icon from '../shared/icons/Icon';
+import {DarkTheme} from '../../themes/DarkTheme';
+import {Variants} from '../shared/constants/variants';
+import {Spacing} from '../shared/constants/spacing';
+import Grid from '@material-ui/core/Grid';
+import Title from '../shared/title/Title';
+
+function ProjectListItem(props) {
+    const [isHovered, setHover] = useState(false);
+    const [disableStateApply, setDisableStateApply] = useState(false);
+    const applySnapshot = () => {
+        if (disableStateApply) {
+            return;
+        }
+        if (props.onStateSnapshotApply) props.onStateSnapshotApply();
+        setDisableStateApply(true);
+        props.state.applyState(props.project.snapshot).then(() => {
+            setDisableStateApply(false);
+        });
+    }
+
+    return (
+        <Root key={index} spacing={Spacing.Lite}  onClick={() => applySnapshot()} onMouseEnter={() => setHover(true)} onMouseLeave={() => setHover(false)}>
```

Comment:
Since we've created a <ProjectListItem> component, which always has a child icon, we can use CSS to style nested components on :hover instead of using state.